pkg/proxy/proxy.go

```diff
@@ -42,21 +49,26 @@ func (proxy *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (proxy *Proxy) Http(w http.ResponseWriter, r *http.Request) {
-	resp, err := proxy.Client.Transport.RoundTrip(r)
-	if err != nil {
-		log.Warnf("HTTP round trip: %s", err)
-		w.WriteHeader(http.StatusBadGateway)
-		return
-	}
-	defer resp.Body.Close()
+	if (r.Method == http.MethodGet && r.URL.Host == "" && r.URL.Path == "/pac") {
+		w.Write([]byte(proxy.PAC.Source))
+		w.Write([]byte("\n"))
+	} else {
+		resp, err := proxy.Client.Transport.RoundTrip(r)
+		if err != nil {
+			log.Warnf("HTTP round trip: %s", err)
+			w.WriteHeader(http.StatusBadGateway)
+			return
+		}
+		defer resp.Body.Close()
 
-	for name, values := range resp.Header {
-		w.Header()[name] = values
-	}
+		for name, values := range resp.Header {
+			w.Header()[name] = values
+		}
 
-	w.WriteHeader(resp.StatusCode)
+		w.WriteHeader(resp.StatusCode)
 
-	io.Copy(w, resp.Body)
+		io.Copy(w, resp.Body)
+	}
```

Comment:
This package meant to be used as a library, so I'd like to find a way to support serving the PAC file without adding any special cases to this.

I think it should be possible to change just the [HTTP server in `main.go`](https://github.com/robbiemcmichael/paxy/blob/b0a915bc81b15a2c04aa3c4a038a0c23f4642e91/main.go#L59) which is just used by the `paxy` executable. If we write a handler function that checks if `r.URL.Host == ""` then we should be able to send it to a special case handler that can server the PAC file.